src/provider.c

```diff
@@ -1059,6 +1059,16 @@ static int p11prov_module_init(P11PROV_CTX *ctx)
         ctx->module = strdup(env_module);
     }
 
+    /* If the module is not specified in the configuration file, use the p11-kit proxy  */
+    if (ctx->module == NULL) {
+#ifdef DEFAULT_PKCS11_MODULE
+        ctx->module = strdup(DEFAULT_PKCS11_MODULE);
+#else
+        P11PROV_debug("No PKCS#11 module specified.");
```

Comment:
Perhaps this should be a P11PROV_raise() so that we return a meaningful error to the openssl stack as well.

Reply:
Good idea. Updated.